src/streams/EditStreamCard.js

```diff
@@ -75,10 +90,14 @@ export default class EditStreamCard extends PureComponent<Props, State> {
         <ZulipButton
           style={styles.marginTop}
           text={isNewStream ? 'Create' : 'Update'}
-          disabled={name.length === 0}
+          disabled={canUpdateStream}
```

Comment:
You meant to do a negative here, right?
`disabled={!canUpdateStream}`

And probably the logic would need changing then?

Reply:
Right. Sorry. I force pushed a new version.